python/ray/data/tests/test_dataset.py

```diff
@@ -578,6 +579,21 @@ def mapper(col_arrs):
             {"a": 4, "b": 7.0, "c": np.array([6, 7])},
         ],
     )
+    # Single column (should still yield ndarray dict batches).
+    t = t.select(["c"])
+    res = (
+        ray.data.from_arrow(t)
+        .map_batches(mapper, batch_size=2, batch_format="numpy")
```

Comment:
Maybe I'm being really dumb, but how is `map_batches` not raising an assertion error? `mapper` asserts that 

```
        assert list(col_arrs.keys()) == ["a", "b", "c"]
```

but the input table only contains a "c" column?

Reply:
Not being dumb, that's a good point! I'm not sure how these tests passed locally for me... I didn't finish porting the tests.

Let me dig into this.